Editor: Fix Editor Notice to open in another tab on command + click

jeremeylduvall opened the pull request:
Per the comment [here](https://github.com/Automattic/wp-calypso/pull/12114#issuecomment-290069200), we're removing the `visible` icon if the Editor Notice will open the in-app preview.

This also updates how we display the preview to use `onPreview` instead of `iframePreview`. The former [already has logic in place](https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/post-editor.jsx#L587) to detect command + click and open the preview in a new window as a result.

**To test**
1. Open a new post in the Editor
2. Publish the post. You should see the new notice without the `visible` icon
3. Command + click the action in the notice. It should open in a new tab
4. If you update the post again and click the action normally, you should see the preview

**Before**
<img width="294" alt="screen shot 2017-03-30 at 6 16 53 am" src="https://cloud.githubusercontent.com/assets/7240478/24503705/7dbf1ce8-1510-11e7-8d31-2c22e45c9132.png">

**After**
<img width="547" alt="screen shot 2017-03-30 at 6 12 44 am" src="https://cloud.githubusercontent.com/assets/7240478/24503688/66bf9838-1510-11e7-94e3-fba36058f027.png">

folletto commented:
Thanks for this! :)

Design wise, works well — 👍 

> The former already has logic in place to detect command + click and open the preview in a new window as a result.

This is an improvement on not doing anything, so good. 
However I've to note that it doesn't solve the mouse middle-click, which should open a new tab too. I'm not asking it to be fixed here as it would be out of scope (unless you know how to do that quickly) just noting for the future.

folletto commented:
Code looks good too :shipit:

jeremeylduvall commented:
Thanks! Let me fix the tests. In working on something else just now, I also noticed that the preview action is appearing on draft failed messages.

<img width="1001" alt="screen shot 2017-03-30 at 6 30 54 am" src="https://cloud.githubusercontent.com/assets/7240478/24504250/c351fd28-1512-11e7-944f-a9944bff45ff.png">

That's a regression from when this was introduced. Should be a quick fix. Will get that done in here as well.

jeremeylduvall commented:
I updated the conditional for `NoticeAction` to check for `onViewClick && isSitePreviewable && link`. We only provide `link` [on success](https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/post-editor.jsx#L775) so although it's not required to render `NoticeAction` with the in-app preview, it's a quick way to determine if `NoticeAction` should be rendered and hide it on failures.

I also updated the test cases for the editor notice to catch this behavior moving forward. @folletto Can I get one last sanity check?

folletto commented:
Checked both success and failure, looks good — 👍 

Visual reference:

![screen shot 2017-03-30 at 17 40 35](https://cloud.githubusercontent.com/assets/4389/24515605/07f8f14a-1570-11e7-945e-6ed91ccff883.png)
